src/lib/objects/split/index.ts

```diff
@@ -0,0 +1,98 @@
+import type { WakuObjectSvelteDescriptor } from '..'
+import ChatComponent from './chat.svelte'
+import { DataMessageSchema, type DataMessage, type Store, type Balance } from './schemas'
+import StandaloneComponent from './standalone.svelte'
+import logo from './logo.svg'
+import type { View } from './types'
+import { getBalances } from './blockchain'
+import { calculateAmountOwed } from './utils'
+
+export const SPLIT_OBJECT_ID = 'split'
+
+export const splitDescriptor: WakuObjectSvelteDescriptor<Store, DataMessage, View> = {
+	objectId: SPLIT_OBJECT_ID,
+	name: 'Split',
+	description: 'Share expenses within a group',
+	logo,
+
+	wakuObject: ChatComponent,
+
+	standalone: StandaloneComponent,
+
+	onMessage: async (message, args) => {
+		const res = DataMessageSchema.safeParse(message.data)
+		if (!res.success) {
+			console.error(res.error)
+			return
+		}
+
+		if (res.data.type === 'expense') {
+			const { expense, splitterAddress } = res.data
+
+			let balances: Balance[] = []
+			try {
+				balances = await getBalances(args.getContract, splitterAddress)
```

Comment:
Under what circumstances this may fail? This looks like the error is used as a regular conditional, but I suspect that it may have multiple reasons and maybe it would make sense to more clearly differentiate. For example if the blockchain connection fails maybe we should not calculate things and update the store.

Reply:
You are right and I actually wanted to discuss this with you. It may fail because the contract address is wrong or because the blockchain RPC connection fails. Normally I would just leave the `balances = []` but I already had the offchain function to calculate ready.

I guess we should have a broader discussion about error handling. Not only blockchain error but also waku errors or maybe even attacks. For example sometimes even object messages arrives twice e.g. new expense which happened few times during development but no idea how to reproduce. I'll create an issue about it.